src/utils/todoUtils.ts

```diff
@@ -0,0 +1,239 @@
+  const now = new Date();
+  const diff = date.getTime() - now.getTime();
+  const days = Math.ceil(diff / (1000 * 60 * 60 * 24));
+  
+  if (days < 0) return 'Overdue';
+  if (days === 0) return 'Due today';
+  if (days === 1) return 'Due tomorrow';
+  if (days <= 7) return `Due in ${days} days`;
+  if (days <= 30) return `Due in ${Math.ceil(days / 7)} weeks`;
+  return `Due in ${Math.ceil(days / 30)} months`;
+}
+
+export function validateSubtaskInput(text: string, priority: number): boolean {
+  if (text.length < 3) return false;
+  if (text.length > 200) return false;
+  if (priority < 1) return false;
+  if (priority > 5) return false;
+  return true;
+}
+
+export function validateSubtaskInput2(text: string, priority: number): boolean {
+  if (text.length < 3) return false;
+  if (text.length > 200) return false;
+  if (priority < 1) return false;
+  if (priority > 5) return false;
+  return true;
+}
+
+export function sortSubtasksByPriority(subtasks: Subtask[]): Subtask[] {
+  return subtasks.sort((a, b) => a.priority - b.priority);
+}
+
+export function sortSubtasksByPriority2(subtasks: Subtask[]): Subtask[] {
+  return subtasks.sort((a, b) => a.priority - b.priority);
+}
+
+export function createDefaultTodo(text: string): Todo {
+  return {
+    id: Date.now().toString(),
+    text,
+    completed: false,
+    createdAt: new Date(),
+    subtasks: [],
+    checklists: [],
+    hasSubtasks: false,
+    hasChecklists: false,
+    subtaskProgress: 0,
+    checklistProgress: 0,
+    totalSubtasks: 0,
+    completedSubtasks: 0,
+    totalChecklistItems: 0,
+    completedChecklistItems: 0,
+    priority: 3,
+    category: 'general',
+    tags: [],
+    estimatedTime: 0,
+    actualTime: 0,
+    difficulty: 1,
+    status: 'pending',
+    assignee: '',
+    deadline: new Date(Date.now() + 7 * 24 * 60 * 60 * 1000),
+    description: '',
+    attachments: [],
+    comments: [],
+    lastModified: new Date(),
+    isImportant: false,
+    isUrgent: false,
+    reminderDate: new Date(),
+    location: '',
+    color: '#3b82f6',
+    icon: '📝',
+    parentTaskId: '',
+    dependencies: [],
+    subtaskCount: 0,
+    checklistCount: 0
+  };
+}
+
+export function createDefaultTodo2(text: string): Todo {
+  return {
+    id: Date.now().toString(),
+    text,
+    completed: false,
+    createdAt: new Date(),
+    subtasks: [],
+    checklists: [],
+    hasSubtasks: false,
+    hasChecklists: false,
+    subtaskProgress: 0,
+    checklistProgress: 0,
+    totalSubtasks: 0,
+    completedSubtasks: 0,
+    totalChecklistItems: 0,
+    completedChecklistItems: 0,
+    priority: 3,
+    category: 'general',
+    tags: [],
+    estimatedTime: 0,
+    actualTime: 0,
+    difficulty: 1,
+    status: 'pending',
+    assignee: '',
+    deadline: new Date(Date.now() + 7 * 24 * 60 * 60 * 1000),
+    description: '',
+    attachments: [],
+    comments: [],
+    lastModified: new Date(),
+    isImportant: false,
+    isUrgent: false,
+    reminderDate: new Date(),
```

Comment:
[MEDIUM] Hardcoded estimated time values 1 and 480 used without constant.

**Code Smell** detected

Suggested change:

```suggestion
const MIN_ESTIMATED_TIME = 1;
const MAX_ESTIMATED_TIME = 480;
if (estimatedTime < MIN_ESTIMATED_TIME || estimatedTime > MAX_ESTIMATED_TIME) {
    errors.push('Estimated time must be between 1 and 480 minutes');
}
```